src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -44,15 +44,21 @@
 
     private transient final KubernetesCloud cloud;
 
-    @DataBoundConstructor
     public KubernetesSlave(PodTemplate template, String nodeDescription, KubernetesCloud cloud, String labelStr)
             throws Descriptor.FormException, IOException {
+
+        this(template, nodeDescription, cloud, labelStr, new OnceRetentionStrategy(cloud.getRetentionTimeout()));
+    }
+
+    @Deprecated
+    public KubernetesSlave(PodTemplate template, String nodeDescription, KubernetesCloud cloud, Label label)
+            throws Descriptor.FormException, IOException {
```

Comment:
can you just call

    this(template, nodeDescription, cloud, label, new OnceRetentionStrategy(cloud.getRetentionTimeout()))

to avoid duplicated code